core_impl/src/main/java/io/opencensus/implcore/stats/MutableAggregation.java

```diff
@@ -355,76 +214,133 @@ long getCount() {
+      this.bucketBoundaries = bucketBoundaries;
+      this.bucketCounts = new long[bucketBoundaries.getBoundaries().size() + 1];
     }
 
     /**
-     * Construct a {@code MutableStdDev}.
+     * Construct a {@code MutableDistribution}.
      *
-     * @return an empty {@code MutableStdDev}.
+     * @return an empty {@code MutableDistribution}.
      */
-    static MutableStdDev create() {
-      return new MutableStdDev();
+    static MutableDistribution create(BucketBoundaries bucketBoundaries) {
+      checkNotNull(bucketBoundaries, "bucketBoundaries should not be null.");
+      return new MutableDistribution(bucketBoundaries);
     }
 
-    /**
-     * Update the sum of squared deviations from the mean with the given value. For values
-     * x_i this is Sum[i=1..n]((x_i - mean)^2)
-     *
-     * <p>Computed using Welfords method (see
-     * https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance, or Knuth, "The Art of
-     * Computer Programming", Vol. 2, page 323, 3rd edition)
-     */
     @Override
     void add(double value) {
+      sum += value;
       count++;
+
+      /*
+       * Update the sum of squared deviations from the mean with the given value. For values
+       * x_i this is Sum[i=1..n]((x_i - mean)^2)
+       *
+       * Computed using Welfords method (see
+       * https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance, or Knuth, "The Art of
+       * Computer Programming", Vol. 2, page 323, 3rd edition)
+       */
       double deltaFromMean = value - mean;
       mean += deltaFromMean / count;
       double deltaFromMean2 = value - mean;
       sumOfSquaredDeviations += deltaFromMean * deltaFromMean2;
+
+      if (value < min) {
+        min = value;
+      }
+      if (value > max) {
+        max = value;
+      }
+
+      for (int i = 0; i < bucketBoundaries.getBoundaries().size(); i++) {
+        if (value < bucketBoundaries.getBoundaries().get(i)) {
+          bucketCounts[i]++;
+          return;
+        }
+      }
+      bucketCounts[bucketCounts.length - 1]++;
     }
 
+    // We don't compute fractional MutableDistribution, it's either whole or none.
     @Override
     void combine(MutableAggregation other, double fraction) {
-      throw new UnsupportedOperationException("Not supported in v0.1.");
+      checkArgument(other instanceof MutableDistribution, "MutableDistribution expected.");
+      if (fraction != 1) {
+        return;
+      }
+
+      MutableDistribution mutableDistribution = (MutableDistribution) other;
+      checkArgument(this.bucketBoundaries.equals(mutableDistribution.bucketBoundaries),
+          "Bucket boundaries should match.");
+
+      this.sum += mutableDistribution.sum;
+      this.count += mutableDistribution.count;
+      this.mean = this.sum / this.count;
+      this.min = mutableDistribution.min < this.min ? mutableDistribution.min : this.min;
+      this.max = mutableDistribution.max > this.max ? mutableDistribution.max : this.max;
```

Comment:
Suggestion: Setting min and max might be simpler with an if statement, as in add(double).

Reply:
Done.